static/src/javascripts/projects/commercial/modules/sticky-top-banner.js

```diff
@@ -0,0 +1,210 @@
+// @flow
+import type { Advert } from 'commercial/modules/dfp/Advert';
+
+import { addEventListener } from 'lib/events';
+import config from 'lib/config';
+import detect from 'lib/detect';
+import fastdom from 'lib/fastdom-promise';
+import trackAdRender from 'commercial/modules/dfp/track-ad-render';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
+import { getAdvertById } from 'commercial/modules/dfp/get-advert-by-id';
+import { register, unregister } from 'commercial/modules/messenger';
+
+const topSlotId = 'dfp-ad--top-above-nav';
+let updateQueued = false;
+let header: ?HTMLElement;
+let headerHeight;
```

Comment:
In the last couple of days I got afraid by some of the DOM measurement methods, so I'd propose to add type definitions for these aswell, just to make clear whether these are strings or numbers. What do you think?